### DISCUSSION TITLE
Offline use

### DISCUSSION THREAD

**Discussion by seahawk1986** [AnswerDotAI/fasthtml#227] [Ideas]
It would be nice if there was a good way to use FastHTML without dependencies on an internet connection resp. CDNs.

It seems you have to disable `default_hdrs` and then set all headers manually:

```python
from fasthtml import *

# use static urls
htmxscr = Script(src="/static/js/htmx.min.js")
htmxwsscr = Script(src="/static/js/ws.js")
surrsrc   = Script(src="/static/js/surreal.js")
scopesrc  = Script(src="/static/js/script.js")
viewport  = Meta(name="viewport", content="width=device-width, initial-scale=1, viewport-fit=cover")
charset   = Meta(charset="utf-8")

picolink = (Link(rel="stylesheet", href='/static/css/pico.min.css'),
            Style(":root { --pico-font-size: 100%; }"))

routes = [
    Mount('/static', app=StaticFiles(directory='static'), name="static")
]

app = FastHTML(default_hdrs=False, hdrs=(htmxscr, htmxwsscr, surrsrc, scopesrc, picolink), routes=routes)
```
Aside from a script that populates resp. updates the js files in the static folder when needed - is this the how you would do it or is there a better way?

**Comment by thomaspujol69** [comment]
I completely agree with your suggestion.
Today for example, `https://unpkg.com/htmx.org@next/dist/htmx.min.js` responded for at least an hour that `htmx.org@next` didn't exist, instead of redirecting to `https://unpkg.com/htmx.org@2.0.3/dist/htmx.min.js`, which led me to manually download the CDN resources, place them in a `static` folder and executing a `sed` to remove the hardcoded URLs.
I have no opinion on a lasting implementation of this feature, but your solution sounds good.
